lib/AST/ASTMangler.cpp

```diff
@@ -1509,6 +1509,14 @@ void ASTMangler::appendType(Type type, GenericSignature sig,
       appendOperator("XSa");
       return;
 
+    case TypeKind::InlineArray: {
+      assert(DWARFMangling && "sugared types are only legal for the debugger");
+      // FIXME: Once we have a known mangling substitution for InlineArray,
+      // use that. For now just desugar.
```

Comment:
We'll need a mangling for the sugar for the ForDebugger mangling kind.

Reply:
> Is there any harm in mangling it as the desugared type for now?

1. nobody will come and fix this FIXME later, and you have all the necessary context in memory right now
2. A type that is spelled as `[3 x Int]` in source code will show up as an `InlineArray<3, Int>` in the debugger, which is confusing for anyone not familiar with the implementation
3. LLDB data formatters match on type names, so this has an effect on how people need to register their custom formatters. This means it would be higher friction to change this later.

Reply:
You can probably just copy&paste the Array sugar mangling (with a different letter)

Reply:
Right, I was hoping we could use the same known-type mangling as InlineArray though (to match the other sugared manglings), which doesn't currently exist. That being said, looking at the known-type manglings, it seems like we've actually exhausted all the available single-letter manglings and would need to add a 3rd level of known-type manglings if we wanted to add InlineArray. So maybe we should just pick an arbitrary letter for the sugared mangling for now (`A` seems fine, seems unlikely `AutoreleasingUnsafeMutablePointer` will ever get a sugar type)